pytorch_lightning/trainer/trainer.py

```diff
@@ -2043,10 +2043,14 @@ def devices(self) -> Optional[Union[List[int], str, int]]:
         return self._accelerator_connector.devices
 
     @property
-    def data_parallel_device_ids(self) -> Optional[List[int]]:
-        return (
-            self._accelerator_connector.parallel_device_ids if self._accelerator_connector.parallel_device_ids else None
-        )
+    def data_parallel_device_ids(self) -> List[int]:
+        rank_zero_deprecation("`Trainer.data_parallel_device_ids` was deprecated in v1.6 and will be removed in v1.8.")
+        if isinstance(self.accelerator, GPUAccelerator):
+            if isinstance(self.strategy, ParallelStrategy):
+                return [device.index for device in self.strategy.parallel_devices]
+            elif isinstance(self.strategy, SingleDeviceStrategy):
+                return [0]
```

Comment:
why is this hardcoded to 0? someone might be running on a different device index on a host. 

use this instead on the strategy's `root_device`
https://github.com/PyTorchLightning/pytorch-lightning/blob/1b107c58928e75b42155aef4a3695b5ee243c276/pytorch_lightning/accelerators/gpu.py#L117

Reply:
Thanks for the suggestion, addressed it. I also added one test case `tests/utilities/test_cli.py::test_parse_args_parsing_gpus` to make sure it returns the right index on SingleDeviceStrategy